internal/agent/inventory.go

```diff
@@ -28,29 +25,22 @@ type InventoryData struct {
 	Error     string        `json:"error"`
 }
 
-func NewInventoryUpdater(log *log.PrefixLogger, config *Config, credUrl string, client client.Planner) *InventoryUpdater {
-	return &InventoryUpdater{
+func NewInventoryUpdater(log *log.PrefixLogger, agentID uuid.UUID, client client.Planner) *InventoryUpdater {
+	updater := &InventoryUpdater{
 		log:        log,
-		config:     config,
 		client:     client,
+		agentID:    agentID,
 		prevStatus: []byte{},
-		credUrl:    credUrl,
 	}
+	return updater
 }
 
-func (u *InventoryUpdater) UpdateServiceWithInventory(ctx context.Context) {
-	status, statusInfo, inventory := calculateStatus(u.config.DataDir)
-	u.updateSourceStatus(ctx, status, statusInfo, inventory)
-}
-
-func (u *InventoryUpdater) updateSourceStatus(ctx context.Context, status api.SourceStatus, statusInfo string, inventory *api.Inventory) {
+func (u *InventoryUpdater) UpdateServiceWithInventory(ctx context.Context, sourceID uuid.UUID, status api.SourceStatus, statusInfo string, inventory *api.Inventory) {
 	update := agentapi.SourceStatusUpdate{
-		Status:        string(status),
-		StatusInfo:    statusInfo,
-		Inventory:     inventory,
-		CredentialUrl: u.credUrl,
-		// TODO: when moving to AgentStatusUpdate put this:
-		//Version: version,
+		Status:     string(status),
+		StatusInfo: statusInfo,
+		Inventory:  inventory,
+		AgentId:    u.agentID,
```

Comment:
Please add `Version: version`

Reply:
not here. this is the source. the version is passed in the agent update method